qiskit/chemistry/drivers/integrals_driver.py

```diff
@@ -33,15 +34,11 @@ class HFMethodType(Enum):
     UHF = 'uhf'
 
 
-class BaseDriver(ABC):
+class IntegralsDriver(BaseDriver):
```

Comment:
Work was still needed - I did not know it was going to be PR'd before things like the GaussianLogDriver was sorted. Either way it was wanted that any driver have the same interface so for things like the GroundStateCalc it can be done for a Bosonic or Fermionic driver the same way so is typed to a BaseDriver. Given they are not the same, the outputs are different and hence arguably the recipient will have to actually check type compatibility to do that the intent was to tag them. I suggested IntegralsDriver (as a case of a FermionicDriver) my thought on GaussianLogDriver was that it would be a ForcesDriver. Since there could be other possible drivers I was reluctant to call them Fermionic and Bosonic, though if we really needed to tag them with that too it could be done.

Then the goal is to use a driver the same way once its created. Whether with a molecule as input or some other config/file. Now 4 of the drivers can be given a molecule instead of their specific config. A getter would allow the molecule to be accessed (would return None perhaps if created from device specific config) and its geometry perturbed. The molecule is re-read each time the driver is run so simply perturbing the molecule and re-rerunning the transformation gets the updated operators. I would argue that given you need to query whether the driver has a molecule to perturb it etc that the so-called parser driver type is not needed rather they will always return None for the molecule.

Now I agree that the constructor becomes a little more complex but it allows, on PySCF and PyQuante to also use the driver specific fields when the user creates the driver. We went with this as a choice on the constructor to keep the same number of drivers and allow the advanced settings to be used for some of them. The other thought was a MoleculeDriver(molecule, basis-set, method, 'auto'|driver-name) where it was just the generic info that was passed and internally this would create and instance of an actual driver where 'auto' would find one and say 'pyscf' would use the pyscf one. Here there would be no access to advance config but the constructor is more obvious.

I am happy to have an offline discussion if it helps - I discussed briefly with Panos in some other meeting today. Either way this PR needs more work before its ready but the above outlines the intent. I think it more or less aligns with the notions of types you have but the main aspect is really how to bring the generic molecule capability in and still let the drivers be consumed the same way etc.

Reply:
Thank you for your detailed explanation! :) I also think that an offline discussion would be beneficial.

Yesterday, I talked with @paulineollitrault  about how we will integrate the vibrational structure calculations with this PR and #1288. Thus, it is probably a good idea to discuss all of this together next Monday.

Reply:
```
                                  BaseDriver

     FermionicDriver                MoleculeDriver              BosonicDriver
     +run() -> QMolecule            +from_molecule()            +run() -> "super-class" of GaussianLogResult

     FCIDumpDriver          PySCFDriver                         VibronicGaussianDriver (prev. GaussianLogDriver)
     HDF5Driver             Psi4Driver
                            PyQuanteDriver
                            GaussianDriver                                   [detailed structure TBD]
```